### ISSUE TITLE
Creating testdriven-app_users_1    ... error

### ISSUE THREAD

**Issue by vbandelier** [testdrivenio/testdriven-app-2.3#17]
Hi Michael,

I have a problem with your tutorial in part 1.

In fact, this error occurs when I run the command `docker-compose -f docker-compose-dev.yml up -d --build` of page 28 :

```
testdriven-app_users-db_1 is up-to-date
Starting testdriven-app_users_1 ... error

ERROR: for testdriven-app_users_1  Cannot start service users: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"/usr/src/app/entrypoint.sh\": permission denied": unknown

ERROR: for users  Cannot start service users: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"/usr/src/app/entrypoint.sh\": permission denied": unknown
ERROR: Encountered errors while bringing up the project.
```

Can you help me on this point please ?

Thank you very much and have a nice day.

Valentin

**Comment by kjuulh** [comment]
Try running
`chmod +x entrypoint.sh`
in the users folder. It seems to be a permission problem.

**Comment by vbandelier** [comment]
It's working! Thank you very much!

**Comment by mjhea0** [comment]
I will update this in the dockerfiles for the next release (2.3.1) in the next few weeks. Thanks!

**Comment by kjuulh** [comment]
Btw this might happen on windows as well running on Linux.

It is very important that the files have permissions in order. Otherwise, massive headaches might occur. I develop on windows which have a not so clear permissions setting - that doesn't register in GitHub. Therefore it is very important that as an example Entrypoint has both Read and Execute permissions otherwise op's situation might occur, A quick fix is to take it to the terminal, go to the root (of the project) and shove a big pile of `chmod -R 777 .` (this will make all the files read, write and executable recursively, this can potentially be dangerous) otherwise only chmod the files that actually needs the permissions. 

I fixed it with using an Ubuntu 18.04 subsystem on windows cloned the GitHub, changed the permissions and pushed back up. Ubuntu subsystem is available in the windows store. 

I can see that this is a general issue across the board of all the versions on this book, so I strongly recommend adding a section to illustrate that permission is a vital part of following along with this book.

I haven't tested it yet, but user-level permissions might be set from an application that is run from as non-admin, and those are not available to the Travis or docker-machine interface.

I hope this helped those developing on windows, otherwise, you can contact me directly and we can figure something out...

**Comment by ericcgu** [comment]
An executable file needs to have permissions for execute set before you can execute it.

In your machine where you are building the docker image (not inside the docker image itself) try running:

ls -la path/to/directory
The first column of the output for your executable (in this case docker-entrypoint.sh) should have the executable bits set something like:

-rwxrwxr-x
If not then try:

chmod +x docker-entrypoint.sh
and then build your docker image again.

Docker uses it's own file system but it copies everything over (including permissions bits) from the source directories.
